File: next-js-boilerplate/src/api/client/profile/actions.ts
export function useProfileActions() {
  const updateProfile = async (data: {
    name?: string;
    username?: string;
    bio?: string;
    avatarUrl?: string;
    chatNickname?: string | null;
  }) => {
    const { updateProfileServer } = await import("@/api/server/profile/update");
    await updateProfileServer(data);
  };

  const uploadAvatar = async (file: File) => {
    const { uploadAvatarServer } =
      await import("@/api/server/profile/upload-avatar");
    return uploadAvatarServer(file);
  };

  const checkUsername = async (username: string) => {
    const { checkUsernameAvailableServer } =
      await import("@/api/server/profile/username-available");
    return checkUsernameAvailableServer(username);
  };

  return { updateProfile, uploadAvatar, checkUsername };
}

File: flutter-boilerplate/lib/api/client/profile/actions.dart
import 'package:flutter_riverpod/flutter_riverpod.dart';

import '../../server/profile/update.dart';
import '../../server/profile/upload_avatar.dart';
import '../../server/profile/username_available.dart';

final profileActionsProvider = Provider((ref) => ProfileActions(ref));

class ProfileActions {
  final Ref _ref;

  ProfileActions(this._ref);

  Future<void> update({
    String? name,
    String? bio,
    String? username,
    String? avatarUrl,
    String? locale,
    String? timezone,
    String? chatNickname,
    bool? useNickname,
    bool? hideAvatar,
  }) async {
    final server = _ref.read(profileUpdateServerProvider);
    await server.call(
      name: name,
      bio: bio,
      username: username,
      avatarUrl: avatarUrl,
      locale: locale,
      timezone: timezone,
      chatNickname: chatNickname,
      useNickname: useNickname,
      hideAvatar: hideAvatar,
    );
  }

  Future<String> uploadAvatar(String filePath) async {
    final server = _ref.read(profileUploadAvatarServerProvider);
    return server.call(filePath);
  }

  Future<bool> checkUsername(String username) async {
    final server = _ref.read(usernameAvailableServerProvider);
    return server.call(username);
  }
}
